content/intro-to-storybook/react/en/data.md

```diff
@@ -105,24 +106,39 @@ PureTaskList.defaultProps = {
   loading: false,
 };
 
-export default connect(
-  ({ tasks }) => ({
-    tasks: tasks.filter(t => t.state === 'TASK_INBOX' || t.state === 'TASK_PINNED'),
-  }),
-  dispatch => ({
-    onArchiveTask: id => dispatch(archiveTask(id)),
-    onPinTask: id => dispatch(pinTask(id)),
-  })
-)(PureTaskList);
+export function TaskList() {
+  // We're retrieving our state from the store
+  const tasks = useSelector(state => state.tasks);
+  // We're defining an variable to handle dispatching the actions back to the store
+  const dispatch = useDispatch();
+
+  const pinTask = value => {
+    // We're dispatching the Pinned event back to our store
+    dispatch(updateTaskState({ id: value, newTaskState: 'TASK_PINNED' }));
+  };
+  const archiveTask = value => {
+    // We're dispatching the Archive event back to our store
+    dispatch(updateTaskState({ id: value, newTaskState: 'TASK_ARCHIVED' }));
+  };
+
+  const filteredTasks = tasks.filter(t => t.state === 'TASK_INBOX' || t.state === 'TASK_PINNED');
+  return (
+    <PureTaskList
+      tasks={filteredTasks}
+      onPinTask={task => pinTask(task)}
```

Comment:
`onPinTask={task => pinTask(task)}` = `onPinTask={pinTask}`.

Is there a reason we wouldn't prefer the simpler form?

Reply:
I'm ok for leaving this for now as this is on its "last legs" (pardon the bad pun) as I've mentioned earlier.